File: js/Chance/chance.js
class Chance {
    constructor(numberOfSuccesses, numberOfTotalOutcomes) {
        this.successes = numberOfSuccesses;
        this.totalOutcomes = numberOfTotalOutcomes;
        
    }

    equals(compareTo) {
        return this.successes / this.totalOutcomes === compareTo.successes / compareTo.totalOutcomes
    }

    invert() {
        let unsuccessfulOutcomes = this.totalOutcomes - this.successes;
        return new Chance(unsuccessfulOutcomes, this.totalOutcomes);
    }
}

module.exports = Chance

File: c#/Chance/Chance/Chance.cs
namespace CleanCode.Chance;

public class Chance
{
    private readonly int successes = 0;
    private readonly int totalOutcomes = 0;

    public Chance(int numberOfSuccesses, int numberOfTotalOutcomes)
    {
        successes = numberOfSuccesses;
        totalOutcomes = numberOfTotalOutcomes;
    }

    public bool Equals(Chance compareTo) {
        var thisLikelihood = (decimal)successes / totalOutcomes;
        var compareLikelihood = (decimal)compareTo.successes / compareTo.totalOutcomes;

        return thisLikelihood == compareLikelihood;
    }
}
